src/client/content_cao.cpp

```diff
@@ -580,6 +602,13 @@ void GenericCAO::addToScene(ITextureSource *tsrc)
 	video::E_MATERIAL_TYPE material_type = (m_prop.use_texture_alpha) ?
 		video::EMT_TRANSPARENT_ALPHA_CHANNEL : video::EMT_TRANSPARENT_ALPHA_CHANNEL_REF;
 
+	IShaderSource *shdrsrc = m_client->getShaderSource();
+
+	if (m_enable_shaders) {
+		u32 shader_id = shdrsrc->getShader("model_shader", TILE_MATERIAL_BASIC, NDT_NORMAL);
+		m_material_type = shdrsrc->getShaderInfo(shader_id).material;
```

Comment:
is it correct that `m_material_type` is only for shaders? it should be renamed too then

Reply:
also in the original PR, model_shader was applied not only to meshes, but to all types of props in addToScene. Not sure if that's correct. Maybe you have any advice on this?

Reply:
Apply model_shader to all kinds of active objects sounds correct to me.

Reply:
Probably I should rename it to "object_shader" then.